docs/mvvm/MigratingFromMvvmLight.md

```diff
@@ -0,0 +1,842 @@
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+```
+
+#### Register<TMessage> ( object, object, bool, Action<TMessage> )
+
+There is no direct replacement for this registration mechanism which allows you to support receiving messages for derived message types also. This change is intentional as the `Messenger` implementation aims to not use reflection to achieve its performance benefits.
+
+Alternatively, there are a few options that can be done to achieve this functionality. 
+
+- Create a custom `IMessenger` implementation
+- Register the additional message types using a shared `Action`
+
+```csharp
+// MvvmLight
+Messenger.Default.Register<MyMessage>(this, nameof(MyViewModel), true, this.OnMyMessageReceived);
+
+// Toolkit.Mvvm
+Messenger.Default.Register<MyMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+Messenger.Default.Register<MyOtherMessage, string>(this, nameof(MyViewModel), this.OnMyMessageReceived);
+```
+
+#### Send<TMessage> ( TMessage )
+
+The functionality of `Send<TMessage>(TMessage)` can be achieved with the MVVM Toolkit's `IMessenger` extension method `Send<TMessage>(TMessage)`. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Send<MyMessage>(new MyMessage());
+Messenger.Default.Send(new MyMessage());
+
+// Toolkit.Mvvm
+Messenger.Default.Send<MyMessage>(new MyMessage());
+Messenger.Default.Send(new MyMessage());
+```
+
+In the above scenario where the message being sent has a parameterless constructor, the MVVM Toolkit has a simplified extension to send a message in this format.
+
+```csharp
+// Toolkit.Mvvm
+Messenger.Default.Send<MyMessage>();
+```
+
+#### Send<TMessage> ( TMessage, object )
+
+The functionality of `Send<TMessage>(TMessage, object)` can be achieved with the MVVM Toolkit's `Send<TMessage, TToken>(TMessage, TToken)` method. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Send<MyMessage>(new MyMessage(), nameof(MyViewModel));
+Messenger.Default.Send(new MyMessage(), nameof(MyViewModel));
+
+// Toolkit.Mvvm
+Messenger.Default.Send(new MyMessage(), nameof(MyViewModel));
+```
+
+#### Unregister ( object )
+
+The functionality of `Unregister(object)` can be achieved with the MVVM Toolkit's `UnregisterAll(object)` method. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Unregister(this);
+
+// Toolkit.Mvvm
+Messenger.Default.UnregisterAll(this);
+```
+
+#### Unregister<TMessage> ( object )
+
+The functionality of `Unregister<TMessage>(object)` can be achieved with the MVVM Toolkit's `IMessenger` extension method `Unregister<TMessage>(object)`. 
+
+```csharp
+// MvvmLight
+Messenger.Default.Unregister<MyMessage>(this);
+
+// Toolkit.Mvvm
+Messenger.Default.Unregister<MyMessage>(this);
```

Comment:
We need to update all these `Messenger.Default`, as this is no longer valid.
It's either just `Messenger` if from within `ObservableRecipient`, or one of the two messenger types 🙂